pkg/scheduler/framework/plugins/noderesources/least_allocated_test.go

```diff
@@ -341,47 +337,59 @@ func TestNodeResourcesLeastAllocated(t *testing.T) {
 			// So the final scores are:
 			// - node1: (50 + 50 + 50) / 3 = 50
 			// - node2: (50 + 50 + 80) / 3 = 60
-			pod: &v1.Pod{Spec: cpuMemoryAndExtendedRes},
+			name:         "honor extended resource if the pod requests",
+			requestedPod: makePodWithExtendedResource("", 3000, 5000, map[string]int64{extendedRes: 2}),
 			nodes: []*v1.Node{
-				makeNodeWithExtendedResource("machine1", 6000, 10000, map[string]int64{extendedRes: 4}),
-				makeNodeWithExtendedResource("machine2", 6000, 10000, map[string]int64{extendedRes: 10}),
+				makeNodeWithExtendedResource("node1", 6000, 10000, map[string]int64{extendedRes: 4}),
+				makeNodeWithExtendedResource("node2", 6000, 10000, map[string]int64{extendedRes: 10}),
+			},
+			scheduledPods:      nil,
+			expectedPriorities: []framework.NodeScore{{Name: "node1", Score: 50}, {Name: "node2", Score: 60}},
+			nodeResourcesFitArgs: config.NodeResourcesFitArgs{
+				ScoringStrategy: &config.ScoringStrategy{
+					Type:      config.LeastAllocated,
+					Resources: extendedResourceLeastAllocatedSet,
+				},
+			},
+			wantErrs: field.ErrorList{
+				&field.Error{
+					Type:  field.ErrorTypeInvalid,
+					Field: "resources[1].weight",
+				},
 			},
-			args:         config.NodeResourcesLeastAllocatedArgs{Resources: extendedResourceLeastAllocatedSet},
-			expectedList: []framework.NodeScore{{Name: "machine1", Score: 50}, {Name: "machine2", Score: 60}},
-			name:         "honor extended resource if the pod requests",
 		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			snapshot := cache.NewSnapshot(test.pods, test.nodes)
+			state := framework.NewCycleState()
+			snapshot := cache.NewSnapshot(test.scheduledPods, test.nodes)
 			fh, _ := runtime.NewFramework(nil, nil, runtime.WithSnapshotSharedLister(snapshot))
-			p, err := NewLeastAllocated(&test.args, fh, feature.Features{EnablePodOverhead: true})
-
-			if test.wantErr != nil {
-				if err != nil {
-					diff := cmp.Diff(test.wantErr, err, cmpopts.IgnoreFields(field.Error{}, "BadValue", "Detail"))
+			args := test.nodeResourcesFitArgs
+			p, err := NewFit(&args, fh, plfeature.Features{EnablePodOverhead: true})
+			if err != nil {
```

Comment:
What if `err` is nil, while we expect an err?

This may be hypothetical but can catch potential bugs of `NewFit()`.

Reply:
good catch. 👍